torch/optim/adadelta.py

```diff
@@ -197,10 +206,16 @@ def _single_tensor_adadelta(params: List[Tensor],
             acc_delta = torch.view_as_real(acc_delta)
             grad = torch.view_as_real(grad)
 
+
         square_avg.mul_(rho).addcmul_(grad, grad, value=1 - rho)
         std = square_avg.add(eps).sqrt_()
-        delta = acc_delta.add(eps).sqrt_().div_(std).mul_(grad)
+        delta = acc_delta.add(eps).sqrt_()
+        if differentiable:
+            std = std.clone()
```

Comment:
Why do you clone std here? It's not modified inplace after this point.

Reply:
right, I was in autopilot mode lol